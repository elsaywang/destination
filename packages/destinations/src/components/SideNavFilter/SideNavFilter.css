:local(.filterType) {
    padding: 1rem 0 0 0;
    font-size: 12px;
}

:local(.link) {
    color: inherit;
    text-decoration: none;
}
