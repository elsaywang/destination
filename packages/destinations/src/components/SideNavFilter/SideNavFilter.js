import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { NavLink, withRouter } from 'react-router-dom';
import { Tab, TabList } from '@react/react-spectrum/TabList';
import styles from './SideNavFilter.css';

const SideNavFilter = ({ filterOptions, filterType, onFilterTypeChange, baseRoute }) => (
    <div className={styles.filterType} data-test="side-nav-filter">
        <TabList
            orientation="vertical"
            variant="compact"
            quiet
            selectedIndex={filterOptions.findIndex(
                ({ value, isDefault }) =>
                    (_.isEmpty(filterType) && isDefault) || value === filterType,
            )}
            onChange={tabIndex => onFilterTypeChange(filterOptions[tabIndex])}>
            {filterOptions.map(({ label, value, subroute }) => (
                <Tab
                    data-test={`${value.toLowerCase().replace(/\W/g, '-')}-type-filter`}
                    key={value}
                    value={value}
                    className={styles.filterType}>
                    <NavLink className={styles.link} to={baseRoute + subroute}>
                        {label}
                    </NavLink>
                </Tab>
            ))}
        </TabList>
    </div>
);

SideNavFilter.propTypes = {
    isSearched: PropTypes.bool,
    onFilterTypeChange: PropTypes.func,
    filterType: PropTypes.string,
    filterOptions: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.string,
            label: PropTypes.string,
        }),
    ),
};

export default withRouter(SideNavFilter);
