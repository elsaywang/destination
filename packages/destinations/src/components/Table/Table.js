import React, { Component } from 'react';
import { TableView } from '@react/react-spectrum/TableView';
import ListDataSource from '@react/react-spectrum/ListDataSource';
import IndexPath from '../../utils/indexPath';
import PropTypes from 'prop-types';

class Table extends Component {
    constructor({ reachedEndOfRows, items = [] }) {
        super();

        this.dataSource = new class DataSource extends ListDataSource {
            load() {
                return items;
            }

            performLoadMore() {
                let totalItems = 0;
                for (let i = 0; i < this.getNumberOfSections(); i++) {
                    totalItems += this.getSectionLength(i);
                }

                reachedEndOfRows(totalItems);
            }
        }();
    }

    componentWillReceiveProps(nextProps) {
        const length = this.props.items.length;

        this.dataSource.insertItems(new IndexPath(0, length), nextProps.items.slice(length));
    }

    render() {
        const { dataTest, styles, height, dataSource, items, ...tableViewProps } = this.props;

        return (
            <div data-test={dataTest} className={styles} style={{ height: `${height}px` }}>
                <TableView dataSource={this.dataSource} {...tableViewProps} />
            </div>
        );
    }
}

Table.defaultProps = {
    items: [],
    rowHeight: 30, // chosen randomly, not sure what a sensible default should be
    allowsSelection: false,
    reachedEndOfRows: () => {
        /* no-op*/
    },
};

Table.protoTypes = {
    height: PropTypes.number,
    columns: PropTypes.array,
    rowHeight: PropTypes.number,
    renderCell: PropTypes.func.isRequired,
    onSelectionChange: PropTypes.func,
    dataTest: PropTypes.string,
    allowsSelection: PropTypes.bool,
    styles: PropTypes.string,
    setRefreshDataSource: PropTypes.func,
    reachedEndOfRows: PropTypes.func,
};

export default Table;
