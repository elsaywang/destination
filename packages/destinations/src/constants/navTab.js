export const routes = [
    { route: '/destinations/all', name: 'All', types: [] },
    {
        route: '/destinations/adobeExperienceCloud',
        name: 'Adobe Experience Cloud',
        types: ['ANALYTICS'],
    },
    { route: '/destinations/integratedPlatforms', name: 'Integrated Platforms', types: ['S2S'] },
    { route: '/destinations/custom', name: 'Custom', types: ['PUSH', 'ADS'] },
];
