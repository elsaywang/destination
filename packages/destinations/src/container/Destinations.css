:local(.destinationContainer){
  display: flex;
  min-height: calc(115vh - var(--top-nav-height) - var(--footer-height) - var(--offset) );
}

:local(.searchText){
  width: 25rem;
}
:local(.filterListContainer) {
    flex: 0 0 10em;
    border-right: 1px solid var(--spectrum-alias-border-color-dark);
}

:local(.tableContainer) {
    flex: 1;
    padding-top: 1rem;
    min-height: 20rem;
}
