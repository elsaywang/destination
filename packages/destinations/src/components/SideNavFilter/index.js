import SideNavFilter from './SideNavFilter';

export default SideNavFilter;
