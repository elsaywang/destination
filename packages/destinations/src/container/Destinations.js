import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Table from '../components/Table';
import SideNavFilter from '../components/SideNavFilter';
import styles from './Destinations.css';
import { integratedPlatformsOptions } from '../constants/integratedPlatformsOptions';
import {
    destinationCategories,
    getCategoryByDestinationType,
    destinationTypeMap,
} from '../constants/destinations';
import {
    updateIntegratedPlatformType,
    applySort,
    applyFilter,
    fetchDestinations,
    fetchMoreDestinations,
    deleteDestination,
} from '../redux/actions/destinations';
import columnsForDestinationType from '../constants/columns';
import Actions from '../components/Actions';

class Destinations extends Component {
    renderCell = (column, data) => {
        const { destinationType } = data;
        const { key } = column;
        switch (key) {
            case 'action':
                return this.renderActionCell(data);
            case 'category':
                return <span>{getCategoryByDestinationType(destinationType)}</span>;
            case 'destinationType':
                return <span>{destinationTypeMap[destinationType]}</span>;
            default:
                return <span>{data[key]}</span>;
        }
    };

    renderActionCell = data => {
        const { currentDestination, deleteDestination } = this.props;
        const { name } = currentDestination;
        const currentRecordCategory = getCategoryByDestinationType(data.destinationType);

        const includeMetrics = this.isIntegratedPlatform(currentRecordCategory);

        return (
            <Actions
                destination={data}
                showMetrics={includeMetrics}
                handleDeleteDestination={deleteDestination}
            />
        );
    };

    isIntegratedPlatform = (category = this.props.currentDestination.name) =>
        category === 'Integrated Platforms';

    sortData = ({ column, direction }) => {
        this.props.applySort({ sortColumn: column, sortDirection: direction });
        this.props.fetchDestinations();
    };

    handleSideNavFilterChange = ({ serverTypes, value }) => {
        this.props.applyFilter(serverTypes);
        this.props.updateIntegratedPlatformType(value);
        this.props.fetchDestinations();
    };

    componentWillMount() {
        this.props.applyFilter(this.props.currentDestination.types);
        this.props.fetchDestinations();
    }

    componentWillUnmount() {
        this.props.updateIntegratedPlatformType('');
    }

    render() {
        const {
            fetchMoreDestinations,
            currentDestination,
            destinations: { integratedPlatformType, byIds, idsToDisplay, replacementDataInFlight },
        } = this.props;

        const renderSideNavFilter = (
            <div className={styles.filterListContainer}>
                <SideNavFilter
                    onFilterTypeChange={this.handleSideNavFilterChange}
                    filterType={integratedPlatformType}
                    isSearched={true}
                    filterOptions={integratedPlatformsOptions}
                />
            </div>
        );

        const destinationsList = idsToDisplay.map(id => byIds[id]);

        return (
            <div
                className={styles.destinationContainer}
                data-test={`${currentDestination.name
                    .toLowerCase()
                    .replace(/\W/g, '-')}-destinations`}>
                {this.isIntegratedPlatform() && renderSideNavFilter}
                <div className={styles.tableContainer}>
                    {replacementDataInFlight ? (
                        <p>Loading</p>
                    ) : (
                        <Table
                            dataTest="destination-list-table"
                            items={destinationsList}
                            onSortChange={this.sortData}
                            reachedEndOfRows={fetchMoreDestinations}
                            sortDescriptor={{
                                column: this.props.destinations.sortColumn,
                                direction: this.props.destinations.sortDirection,
                            }}
                            height={900}
                            columns={
                                columnsForDestinationType[
                                    integratedPlatformType || currentDestination.name
                                ]
                            }
                            rowHeight={250}
                            renderCell={this.renderCell}
                        />
                    )}
                </div>
            </div>
        );
    }
}

Destinations.propTypes = {
    fetchDestinations: PropTypes.func.isRequired,
    destinations: PropTypes.shape({
        byIds: PropTypes.shape({
            id: PropTypes.shape({
                id: PropTypes.number,
                name: PropTypes.string,
                enabled: PropTypes.bool,
                createdBy: PropTypes.string,
            }),
        }),
        idsToDisplay: PropTypes.arrayOf(PropTypes.number),
        integratedPlatformType: PropTypes.string,
        sortColumn: PropTypes.object,
        sortDirection: PropTypes.oneOf([-1, 1]),
    }),
    currentDestination: PropTypes.shape({
        name: PropTypes.oneOf(destinationCategories).isRequired,
        types: PropTypes.array.isRequired,
        route: PropTypes.string.isRequired,
    }),
};

const mapStateToProps = ({ destinations }) => ({
    destinations,
});

const actionCreators = {
    fetchDestinations,
    fetchMoreDestinations,
    updateIntegratedPlatformType,
    deleteDestination,
    applySort,
    applyFilter,
};

export { Destinations };

export default connect(
    mapStateToProps,
    actionCreators,
)(Destinations);
