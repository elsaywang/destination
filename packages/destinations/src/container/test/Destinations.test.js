import React from 'react';
import { shallow } from 'enzyme';
import { Destinations } from '../Destinations';
import Table from '../../components/Table';
import SideNavFilter from '../../components/SideNavFilter';
import { routes } from '../../constants/navTab';

describe('renders <App/> without crashing', () => {
    const noOp = () => {};
    const props = {
        fetchDestinations: noOp,
        fetchMoreDestinations: noOp,
        applyFilter: noOp,
        destinations: {
            byIds: {
                0: {
                    action: '',
                    description: 'Maxime ut totam nemo assumenda ratione impedit.',
                    id: 0,
                    name: 'input',
                    platform: 'Twitter',
                    shareableAudience: 41898,
                },
                1: {
                    action: '',
                    description: 'Et nesciunt voluptatibus facilis blanditiis.',
                    id: 1,
                    name: 'Bedfordshire',
                    platform: 'Google',
                    shareableAudience: 18751,
                },
                2: {
                    action: '',
                    description: 'Repellat sit dolor ut quisquam et ducimus.',
                    id: 2,
                    name: 'Shoes',
                    platform: 'Twitter',
                    shareableAudience: 5434,
                },
                3: {
                    action: '',
                    description: 'Blanditiis tenetur culpa aut distinctio.',
                    id: 3,
                    name: 'mesh',
                    platform: 'LinkedIn',
                    shareableAudience: 40918,
                },
            },
            idsToDisplay: [0, 1, 2, 3],
            replacementDataInFlight: false,
        },
        currentDestination: routes.find(({ name }) => name === 'Integrated Platforms'),
    };
    const wrapper = shallow(<Destinations {...props} />);

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
        it('renders <Table/>', () => {
            expect(wrapper.find(Table).exists()).toBeTruthy();
        });
        it('renders <SideNavFilter/> when `currentDestination` name is `Integrated Platforms`', () => {
            expect(wrapper.find(SideNavFilter).exists()).toBeTruthy();
        });
        it('renders no <SideNavFilter/> when `currentDestination` name is non-`Integrated Platforms`', () => {
            const nonIntegratedPlatform = routes.find(
                ({ name }) => name && name !== 'Integrated Platforms',
            );
            wrapper.setProps({ currentDestination: nonIntegratedPlatform });
            expect(wrapper.find(SideNavFilter).exists()).toBeFalsy();
        });
    });
});
