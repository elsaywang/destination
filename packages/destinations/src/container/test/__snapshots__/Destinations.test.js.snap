// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> without crashing rendering matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Destinations
    destinationType="Integrated Platforms"
    destinations={
        Object {
            "byIds": Object {
              "0": Object {
                "action": "",
                "description": "Maxime ut totam nemo assumenda ratione impedit.",
                "id": 0,
                "name": "input",
                "platform": "Twitter",
                "shareableAudience": 41898,
              },
              "1": Object {
                "action": "",
                "description": "Et nesciunt voluptatibus facilis blanditiis.",
                "id": 1,
                "name": "Bedfordshire",
                "platform": "Google",
                "shareableAudience": 18751,
              },
              "2": Object {
                "action": "",
                "description": "Repellat sit dolor ut quisquam et ducimus.",
                "id": 2,
                "name": "Shoes",
                "platform": "Twitter",
                "shareableAudience": 5434,
              },
              "3": Object {
                "action": "",
                "description": "Blanditiis tenetur culpa aut distinctio.",
                "id": 3,
                "name": "mesh",
                "platform": "LinkedIn",
                "shareableAudience": 40918,
              },
            },
            "idsToDisplay": Array [
              0,
              1,
              2,
              3,
            ],
            "inFlight": false,
          }
    }
    fetchDestinations={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <SideNavFilter
                    filterOptions={
                              Array [
                                        Object {
                                          "label": "ALL PLATFORMS",
                                          "value": "Integrated Platforms",
                                        },
                                        Object {
                                          "label": "PEOPLE-BASED",
                                          "value": "People-Based",
                                        },
                                        Object {
                                          "label": "DEVICE-BASED",
                                          "value": "Device-Based",
                                        },
                                      ]
                    }
                    filterType={undefined}
                    isSearched={true}
                    onFilterTypeChange={[Function]}
          />
</div>,
        <div
          className={undefined}
>
          <Table
                    allowsSelection={false}
                    columns={
                              Array [
                                        Object {
                                          "announce": false,
                                          "key": "id",
                                          "sortable": true,
                                          "title": "ID",
                                          "width": 50,
                                        },
                                        Object {
                                          "active": true,
                                          "key": "type",
                                          "sortable": true,
                                          "title": "TYPE",
                                          "width": 150,
                                        },
                                        Object {
                                          "active": true,
                                          "key": "platform",
                                          "sortable": true,
                                          "title": "PLATFORM",
                                          "width": 150,
                                        },
                                        Object {
                                          "key": "name",
                                          "sortable": true,
                                          "title": "NAME",
                                          "width": 200,
                                        },
                                        Object {
                                          "key": "description",
                                          "minWidth": 250,
                                          "sortable": true,
                                          "title": "DESCRIPTION",
                                        },
                                        Object {
                                          "key": "action",
                                          "title": "ACTION",
                                          "width": 200,
                                        },
                                      ]
                    }
                    dataTest="peopleBased-destination-table"
                    height={900}
                    items={
                              Array [
                                        Object {
                                          "action": "",
                                          "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                          "id": 0,
                                          "name": "input",
                                          "platform": "Twitter",
                                          "shareableAudience": 41898,
                                        },
                                        Object {
                                          "action": "",
                                          "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                          "id": 1,
                                          "name": "Bedfordshire",
                                          "platform": "Google",
                                          "shareableAudience": 18751,
                                        },
                                        Object {
                                          "action": "",
                                          "description": "Repellat sit dolor ut quisquam et ducimus.",
                                          "id": 2,
                                          "name": "Shoes",
                                          "platform": "Twitter",
                                          "shareableAudience": 5434,
                                        },
                                        Object {
                                          "action": "",
                                          "description": "Blanditiis tenetur culpa aut distinctio.",
                                          "id": 3,
                                          "name": "mesh",
                                          "platform": "LinkedIn",
                                          "shareableAudience": 40918,
                                        },
                                      ]
                    }
                    onSortChange={[Function]}
                    reachedEndOfRows={[Function]}
                    renderCell={[Function]}
                    rowHeight={250}
                    sortDescriptor={
                              Object {
                                        "column": Object {
                                          "announce": false,
                                          "key": "id",
                                          "sortable": true,
                                          "title": "ID",
                                          "width": 50,
                                        },
                                        "direction": 1,
                                      }
                    }
          />
</div>,
      ],
      "className": undefined,
      "data-test": "integrated platforms-destinations",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SideNavFilter
            filterOptions={
                        Array [
                                    Object {
                                      "label": "ALL PLATFORMS",
                                      "value": "Integrated Platforms",
                                    },
                                    Object {
                                      "label": "PEOPLE-BASED",
                                      "value": "People-Based",
                                    },
                                    Object {
                                      "label": "DEVICE-BASED",
                                      "value": "Device-Based",
                                    },
                                  ]
            }
            filterType={undefined}
            isSearched={true}
            onFilterTypeChange={[Function]}
/>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterOptions": Array [
              Object {
                "label": "ALL PLATFORMS",
                "value": "Integrated Platforms",
              },
              Object {
                "label": "PEOPLE-BASED",
                "value": "People-Based",
              },
              Object {
                "label": "DEVICE-BASED",
                "value": "Device-Based",
              },
            ],
            "filterType": undefined,
            "isSearched": true,
            "onFilterTypeChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Table
            allowsSelection={false}
            columns={
                        Array [
                                    Object {
                                      "announce": false,
                                      "key": "id",
                                      "sortable": true,
                                      "title": "ID",
                                      "width": 50,
                                    },
                                    Object {
                                      "active": true,
                                      "key": "type",
                                      "sortable": true,
                                      "title": "TYPE",
                                      "width": 150,
                                    },
                                    Object {
                                      "active": true,
                                      "key": "platform",
                                      "sortable": true,
                                      "title": "PLATFORM",
                                      "width": 150,
                                    },
                                    Object {
                                      "key": "name",
                                      "sortable": true,
                                      "title": "NAME",
                                      "width": 200,
                                    },
                                    Object {
                                      "key": "description",
                                      "minWidth": 250,
                                      "sortable": true,
                                      "title": "DESCRIPTION",
                                    },
                                    Object {
                                      "key": "action",
                                      "title": "ACTION",
                                      "width": 200,
                                    },
                                  ]
            }
            dataTest="peopleBased-destination-table"
            height={900}
            items={
                        Array [
                                    Object {
                                      "action": "",
                                      "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                      "id": 0,
                                      "name": "input",
                                      "platform": "Twitter",
                                      "shareableAudience": 41898,
                                    },
                                    Object {
                                      "action": "",
                                      "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                      "id": 1,
                                      "name": "Bedfordshire",
                                      "platform": "Google",
                                      "shareableAudience": 18751,
                                    },
                                    Object {
                                      "action": "",
                                      "description": "Repellat sit dolor ut quisquam et ducimus.",
                                      "id": 2,
                                      "name": "Shoes",
                                      "platform": "Twitter",
                                      "shareableAudience": 5434,
                                    },
                                    Object {
                                      "action": "",
                                      "description": "Blanditiis tenetur culpa aut distinctio.",
                                      "id": 3,
                                      "name": "mesh",
                                      "platform": "LinkedIn",
                                      "shareableAudience": 40918,
                                    },
                                  ]
            }
            onSortChange={[Function]}
            reachedEndOfRows={[Function]}
            renderCell={[Function]}
            rowHeight={250}
            sortDescriptor={
                        Object {
                                    "column": Object {
                                      "announce": false,
                                      "key": "id",
                                      "sortable": true,
                                      "title": "ID",
                                      "width": 50,
                                    },
                                    "direction": 1,
                                  }
            }
/>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowsSelection": false,
            "columns": Array [
              Object {
                "announce": false,
                "key": "id",
                "sortable": true,
                "title": "ID",
                "width": 50,
              },
              Object {
                "active": true,
                "key": "type",
                "sortable": true,
                "title": "TYPE",
                "width": 150,
              },
              Object {
                "active": true,
                "key": "platform",
                "sortable": true,
                "title": "PLATFORM",
                "width": 150,
              },
              Object {
                "key": "name",
                "sortable": true,
                "title": "NAME",
                "width": 200,
              },
              Object {
                "key": "description",
                "minWidth": 250,
                "sortable": true,
                "title": "DESCRIPTION",
              },
              Object {
                "key": "action",
                "title": "ACTION",
                "width": 200,
              },
            ],
            "dataTest": "peopleBased-destination-table",
            "height": 900,
            "items": Array [
              Object {
                "action": "",
                "description": "Maxime ut totam nemo assumenda ratione impedit.",
                "id": 0,
                "name": "input",
                "platform": "Twitter",
                "shareableAudience": 41898,
              },
              Object {
                "action": "",
                "description": "Et nesciunt voluptatibus facilis blanditiis.",
                "id": 1,
                "name": "Bedfordshire",
                "platform": "Google",
                "shareableAudience": 18751,
              },
              Object {
                "action": "",
                "description": "Repellat sit dolor ut quisquam et ducimus.",
                "id": 2,
                "name": "Shoes",
                "platform": "Twitter",
                "shareableAudience": 5434,
              },
              Object {
                "action": "",
                "description": "Blanditiis tenetur culpa aut distinctio.",
                "id": 3,
                "name": "mesh",
                "platform": "LinkedIn",
                "shareableAudience": 40918,
              },
            ],
            "onSortChange": [Function],
            "reachedEndOfRows": [Function],
            "renderCell": [Function],
            "rowHeight": 250,
            "sortDescriptor": Object {
              "column": Object {
                "announce": false,
                "key": "id",
                "sortable": true,
                "title": "ID",
                "width": 50,
              },
              "direction": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <SideNavFilter
                        filterOptions={
                                    Array [
                                                Object {
                                                  "label": "ALL PLATFORMS",
                                                  "value": "Integrated Platforms",
                                                },
                                                Object {
                                                  "label": "PEOPLE-BASED",
                                                  "value": "People-Based",
                                                },
                                                Object {
                                                  "label": "DEVICE-BASED",
                                                  "value": "Device-Based",
                                                },
                                              ]
                        }
                        filterType={undefined}
                        isSearched={true}
                        onFilterTypeChange={[Function]}
            />
</div>,
          <div
            className={undefined}
>
            <Table
                        allowsSelection={false}
                        columns={
                                    Array [
                                                Object {
                                                  "announce": false,
                                                  "key": "id",
                                                  "sortable": true,
                                                  "title": "ID",
                                                  "width": 50,
                                                },
                                                Object {
                                                  "active": true,
                                                  "key": "type",
                                                  "sortable": true,
                                                  "title": "TYPE",
                                                  "width": 150,
                                                },
                                                Object {
                                                  "active": true,
                                                  "key": "platform",
                                                  "sortable": true,
                                                  "title": "PLATFORM",
                                                  "width": 150,
                                                },
                                                Object {
                                                  "key": "name",
                                                  "sortable": true,
                                                  "title": "NAME",
                                                  "width": 200,
                                                },
                                                Object {
                                                  "key": "description",
                                                  "minWidth": 250,
                                                  "sortable": true,
                                                  "title": "DESCRIPTION",
                                                },
                                                Object {
                                                  "key": "action",
                                                  "title": "ACTION",
                                                  "width": 200,
                                                },
                                              ]
                        }
                        dataTest="peopleBased-destination-table"
                        height={900}
                        items={
                                    Array [
                                                Object {
                                                  "action": "",
                                                  "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                                  "id": 0,
                                                  "name": "input",
                                                  "platform": "Twitter",
                                                  "shareableAudience": 41898,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                                  "id": 1,
                                                  "name": "Bedfordshire",
                                                  "platform": "Google",
                                                  "shareableAudience": 18751,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Repellat sit dolor ut quisquam et ducimus.",
                                                  "id": 2,
                                                  "name": "Shoes",
                                                  "platform": "Twitter",
                                                  "shareableAudience": 5434,
                                                },
                                                Object {
                                                  "action": "",
                                                  "description": "Blanditiis tenetur culpa aut distinctio.",
                                                  "id": 3,
                                                  "name": "mesh",
                                                  "platform": "LinkedIn",
                                                  "shareableAudience": 40918,
                                                },
                                              ]
                        }
                        onSortChange={[Function]}
                        reachedEndOfRows={[Function]}
                        renderCell={[Function]}
                        rowHeight={250}
                        sortDescriptor={
                                    Object {
                                                "column": Object {
                                                  "announce": false,
                                                  "key": "id",
                                                  "sortable": true,
                                                  "title": "ID",
                                                  "width": 50,
                                                },
                                                "direction": 1,
                                              }
                        }
            />
</div>,
        ],
        "className": undefined,
        "data-test": "integrated platforms-destinations",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SideNavFilter
              filterOptions={
                            Array [
                                          Object {
                                            "label": "ALL PLATFORMS",
                                            "value": "Integrated Platforms",
                                          },
                                          Object {
                                            "label": "PEOPLE-BASED",
                                            "value": "People-Based",
                                          },
                                          Object {
                                            "label": "DEVICE-BASED",
                                            "value": "Device-Based",
                                          },
                                        ]
              }
              filterType={undefined}
              isSearched={true}
              onFilterTypeChange={[Function]}
/>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "filterOptions": Array [
                Object {
                  "label": "ALL PLATFORMS",
                  "value": "Integrated Platforms",
                },
                Object {
                  "label": "PEOPLE-BASED",
                  "value": "People-Based",
                },
                Object {
                  "label": "DEVICE-BASED",
                  "value": "Device-Based",
                },
              ],
              "filterType": undefined,
              "isSearched": true,
              "onFilterTypeChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Table
              allowsSelection={false}
              columns={
                            Array [
                                          Object {
                                            "announce": false,
                                            "key": "id",
                                            "sortable": true,
                                            "title": "ID",
                                            "width": 50,
                                          },
                                          Object {
                                            "active": true,
                                            "key": "type",
                                            "sortable": true,
                                            "title": "TYPE",
                                            "width": 150,
                                          },
                                          Object {
                                            "active": true,
                                            "key": "platform",
                                            "sortable": true,
                                            "title": "PLATFORM",
                                            "width": 150,
                                          },
                                          Object {
                                            "key": "name",
                                            "sortable": true,
                                            "title": "NAME",
                                            "width": 200,
                                          },
                                          Object {
                                            "key": "description",
                                            "minWidth": 250,
                                            "sortable": true,
                                            "title": "DESCRIPTION",
                                          },
                                          Object {
                                            "key": "action",
                                            "title": "ACTION",
                                            "width": 200,
                                          },
                                        ]
              }
              dataTest="peopleBased-destination-table"
              height={900}
              items={
                            Array [
                                          Object {
                                            "action": "",
                                            "description": "Maxime ut totam nemo assumenda ratione impedit.",
                                            "id": 0,
                                            "name": "input",
                                            "platform": "Twitter",
                                            "shareableAudience": 41898,
                                          },
                                          Object {
                                            "action": "",
                                            "description": "Et nesciunt voluptatibus facilis blanditiis.",
                                            "id": 1,
                                            "name": "Bedfordshire",
                                            "platform": "Google",
                                            "shareableAudience": 18751,
                                          },
                                          Object {
                                            "action": "",
                                            "description": "Repellat sit dolor ut quisquam et ducimus.",
                                            "id": 2,
                                            "name": "Shoes",
                                            "platform": "Twitter",
                                            "shareableAudience": 5434,
                                          },
                                          Object {
                                            "action": "",
                                            "description": "Blanditiis tenetur culpa aut distinctio.",
                                            "id": 3,
                                            "name": "mesh",
                                            "platform": "LinkedIn",
                                            "shareableAudience": 40918,
                                          },
                                        ]
              }
              onSortChange={[Function]}
              reachedEndOfRows={[Function]}
              renderCell={[Function]}
              rowHeight={250}
              sortDescriptor={
                            Object {
                                          "column": Object {
                                            "announce": false,
                                            "key": "id",
                                            "sortable": true,
                                            "title": "ID",
                                            "width": 50,
                                          },
                                          "direction": 1,
                                        }
              }
/>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowsSelection": false,
              "columns": Array [
                Object {
                  "announce": false,
                  "key": "id",
                  "sortable": true,
                  "title": "ID",
                  "width": 50,
                },
                Object {
                  "active": true,
                  "key": "type",
                  "sortable": true,
                  "title": "TYPE",
                  "width": 150,
                },
                Object {
                  "active": true,
                  "key": "platform",
                  "sortable": true,
                  "title": "PLATFORM",
                  "width": 150,
                },
                Object {
                  "key": "name",
                  "sortable": true,
                  "title": "NAME",
                  "width": 200,
                },
                Object {
                  "key": "description",
                  "minWidth": 250,
                  "sortable": true,
                  "title": "DESCRIPTION",
                },
                Object {
                  "key": "action",
                  "title": "ACTION",
                  "width": 200,
                },
              ],
              "dataTest": "peopleBased-destination-table",
              "height": 900,
              "items": Array [
                Object {
                  "action": "",
                  "description": "Maxime ut totam nemo assumenda ratione impedit.",
                  "id": 0,
                  "name": "input",
                  "platform": "Twitter",
                  "shareableAudience": 41898,
                },
                Object {
                  "action": "",
                  "description": "Et nesciunt voluptatibus facilis blanditiis.",
                  "id": 1,
                  "name": "Bedfordshire",
                  "platform": "Google",
                  "shareableAudience": 18751,
                },
                Object {
                  "action": "",
                  "description": "Repellat sit dolor ut quisquam et ducimus.",
                  "id": 2,
                  "name": "Shoes",
                  "platform": "Twitter",
                  "shareableAudience": 5434,
                },
                Object {
                  "action": "",
                  "description": "Blanditiis tenetur culpa aut distinctio.",
                  "id": 3,
                  "name": "mesh",
                  "platform": "LinkedIn",
                  "shareableAudience": 40918,
                },
              ],
              "onSortChange": [Function],
              "reachedEndOfRows": [Function],
              "renderCell": [Function],
              "rowHeight": 250,
              "sortDescriptor": Object {
                "column": Object {
                  "announce": false,
                  "key": "id",
                  "sortable": true,
                  "title": "ID",
                  "width": 50,
                },
                "direction": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
